### ISSUE TITLE
Proxy will thr a excption while handle a proxy-to-proxy connection

### ISSUE THREAD

**Issue by caoli5288** [PaperMC/Waterfall#76]
Proxy will thr a excption while handle a proxy-to-proxy connection since https://github.com/SpigotMC/BungeeCord/commit/504d3c0529e2014d5a1c469206db3bfd440829a7. I think it's not a good idea. As a matter of fact, I'm maintaining a complex network contains bungee-to-bungee.
I also need IP-forwarding over bungee-bungee. I need you help! can you lift this limit? I have given up chat with bungee developers.

**Comment by Janmm14** [comment]
Complex networks could be also set up without bungee-to-bungee connections.
I suggest you set up your own fork or let your developers do that.

**Comment by caoli5288** [comment]
@Janmm14 Some mc server manager by my friend...So we do need bungee-to-bungee connection. I can revert that commit by myself but I do not known how to do ip-forward over bungee-to-bungee.

**Comment by jamierocks** [comment]
Unless you can provide a good use case for bungee to bungee then neither us not bungee will support this.

**Comment by jamierocks** [comment]
It may also help to see https://github.com/SpigotMC/BungeeCord/issues/1939

**Comment by sgdc3** [comment]
About the bungee to bungee: just add a setting to disable the check
About the ip foward: it can be done with a plugin

**Comment by jamierocks** [comment]
As seen in the bungee issue it can cause unexpected errors. We're not going to support something with a probable large amount of changes to upstream without good reason. We're also not going to provide a config setting  for something that more than likely will end up breaking networks.

Tldr we won't provide a config setting unless the support is there, and we won't support without good reason to do so.

**Comment by caoli5288** [comment]
@sgdc3 Any plugin exists?

**Comment by jamierocks** [comment]
Despite request no reasoning was given.

**Comment by Ichbinjoe** [comment]
Any user which wishes to have this functionality and understands the
implications (or does not understand but wants it anyways) can manually
create a patch. It creates a large enough barrier to discourage this type
of networking. Anyone wanting this type of functionality should know there
are far better solutions (granted, custom plugins) but for anyone needing
this, they should be at scale to accommodate

**Comment by jamierocks** [comment]
Myself and Aikar also came to the conclusion that this functionality was up to another fork or even plugin to support.

**Comment by ghost** [comment]
Check it: https://www.spigotmc.org/resources/allowbungeebungeeconnection.28409/
But, it's compatible only with build 1186 of BungeeCord.
Indeed, @md-5 was removed this commit in build 1187: https://github.com/SpigotMC/BungeeCord/commit/6d4ff3caae99c285d00b2302ef7ba5c14416ef32
